experiments/simple_train_step.py

```diff
@@ -0,0 +1,31 @@
+from itertools import chain
+import torch
+
+
+def simple_train_step(optimizer,criterion,device,model,split_val):
+        
+    def train_step(x, y):
+        model.to(device)
+        premise, hypothesis = torch.split(x,split_val,dim=1)
+        
+        first_hidden_pre = torch.rand_like(premise).to(device)
```

Comment:
these 4 lines seem like a default logic that is more related to the model / data than to your training step logic.
You want to keep your training logic as pure as possible, to improve readability and prevent change amplification (in case this default data processing has to change).
Can you think of a more fitting place for this?